compiler/z/codegen/OMRMemoryReference.cpp

```diff
@@ -3032,7 +3032,14 @@ OMR::Z::MemoryReference::generateBinaryEncoding(uint8_t * cursor, TR::CodeGenera
              instructionFormat == TR::Instruction::IsSS1 ||
              instructionFormat == TR::Instruction::IsSS2 ||
              instructionFormat == TR::Instruction::IsSS4 ||
-             instructionFormat == TR::Instruction::IsS)
+             instructionFormat == TR::Instruction::IsS   ||
+             // Vector instruction formats that have 12-bit displacements and no index registers
+             instructionFormat == TR::Instruction::IsVSI ||
+             instructionFormat == TR::Instruction::IsVRV ||
+             instructionFormat == TR::Instruction::IsVRSa||
+             instructionFormat == TR::Instruction::IsVRSb||
+             instructionFormat == TR::Instruction::IsVRSc||
+             instructionFormat == TR::Instruction::IsVRSd)
```

Comment:
@NigelYiboYu did you confirm these are the only formats without an index register?

Reply:
Yes, these are the vector-storage formats (z13 and above) that don't have an index register. Their memory reference are **D(B)**, with D being 12-bits signed.

The only vector-storage format that has **D(X,B)** is VRS format.

Let me check non-vector formats....

Reply:
For non-vector instructions, we missed some trivial formats: **RIS, RRS, SSE and SSF.** I'll add them just in case. However, note that these are either not used in the JIT code base or not used for large jumps in practice. 

There are only 8 RIS and RRS instructions and they are all compare-and-branch instructions such as CIB, CGIB, and etc. They usually don't jump to far.

We have no use of SSE in the JIT.

SSF format consists of 5 instructions, two of which are used: LDP and LGPD.

Reply:
This is certainly an oversight. We should fix all the ones you noted as well. Let's add them to the list.